packages/markdown/components/Code/index.jsx

```diff
@@ -0,0 +1,60 @@
+const React = require('react');
+const PropTypes = require('prop-types');
+
+// Only load CodeMirror in the browser, for SSR
+// apps. Necessary because of people like this:
+// https://github.com/codemirror/CodeMirror/issues/3701#issuecomment-164904534
+const syntaxHighlighter = typeof window !== 'undefined' ? require('@readme/syntax-highlighter') : false;
+const copy = require('copy-to-clipboard');
+
+const remapLang = {
+  c: 'cplusplus',
+  'c++': 'cplusplus',
+  cpp: 'cplusplus',
+  docker: 'dockerfile',
+  html: 'htmlmixed',
+  js: 'javascript',
+  py: 'python',
+  sh: 'shell',
+  bash: 'shell',
+  mysql: 'sql',
+};
+
+function Code(props) {
+  const { className, children, lang, meta } = props;
+
+  const langClass = className.search(/lang(?:uage)?-\w+/) >= 0 ? className.match(/\s?lang(?:uage)?-(\w+)/)[1] : '';
+  const language = lang in remapLang ? remapLang[lang] : lang || langClass;
+
+  return (
+    <React.Fragment>
+      <code className={['rdmd-code', `lang-${language}`].join(' ')} data-lang={language} name={meta}>
+        {!syntaxHighlighter || (
+          <button className="rdmd-code-copy" onClick={() => copy(children[0])}>
```

Comment:
If `children` is empty, will the engine attempt to load this component or will it ignore it?

Reply:
It just render an empty `<pre/>` tag! But because the Unified processor generally normalizes empty blocks (i.e. removes them from the AST object entirely) it rarely reaches the component level without child content.